src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -189,7 +181,7 @@ public override Task<WebSocketReceiveResult> ReceiveAsync(ArraySegment<byte> buf
             {
                 return Task.FromException<WebSocketReceiveResult>(new OperationCanceledException(cancellationToken));
             }
-            ThrowIfDisposed();
+
```

Comment:
shouldn't this whole method occur inside the lock?

Reply:
refactored